server/src/test/java/io/druid/server/AsyncQueryForwardingServletTest.java

```diff
@@ -0,0 +1,195 @@
+import io.druid.server.initialization.JettyServerInitializer;
+import io.druid.server.initialization.BaseJettyTest;
+import io.druid.server.log.RequestLogger;
+import io.druid.server.metrics.NoopServiceEmitter;
+import io.druid.server.router.QueryHostFinder;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.HandlerList;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+
+public class AsyncQueryForwardingServletTest extends BaseJettyTest
+{
+  @Before
+  public void setup() throws Exception
+  {
+    setProperties();
+    Injector injector = setupInjector();
+    final DruidNode node = injector.getInstance(Key.get(DruidNode.class, Self.class));
+    port = node.getPort();
+
+    lifecycle = injector.getInstance(Lifecycle.class);
+    lifecycle.start();
+    ClientHolder holder = injector.getInstance(ClientHolder.class);
+    client = holder.getClient();
+  }
+
+  @Override
+  protected Injector setupInjector()
+  {
+    return Initialization.makeInjectorWithModules(
+        GuiceInjectors.makeStartupInjector(), ImmutableList.<Module>of(
+            new Module()
+            {
+              @Override
+              public void configure(Binder binder)
+              {
+                JsonConfigProvider.bindInstance(
+                    binder, Key.get(DruidNode.class, Self.class), new DruidNode("test", "localhost", null)
+                );
+                binder.bind(JettyServerInitializer.class).to(ProxyJettyServerInit.class).in(LazySingleton.class);
+                Jerseys.addResource(binder, SlowResource.class);
+                Jerseys.addResource(binder, ExceptionResource.class);
+                Jerseys.addResource(binder, DefaultResource.class);
+                LifecycleModule.register(binder, Server.class);
+              }
+            }
+        )
+    );
+  }
+
+  @Test
+  public void testProxyGzipCompression() throws Exception
+  {
+    final URL url = new URL("http://localhost:" + port + "/proxy/default");
+
+    final HttpURLConnection get = (HttpURLConnection) url.openConnection();
+    get.setRequestProperty("Accept-Encoding", "gzip");
+    Assert.assertEquals("gzip", get.getContentEncoding());
+
+    final HttpURLConnection post = (HttpURLConnection) url.openConnection();
+    post.setRequestProperty("Accept-Encoding", "gzip");
+    post.setRequestMethod("POST");
+    Assert.assertEquals("gzip", post.getContentEncoding());
+
+    final HttpURLConnection getNoGzip = (HttpURLConnection) url.openConnection();
+    Assert.assertNotEquals("gzip", getNoGzip.getContentEncoding());
```

Comment:
@xvrl we haven't tested it on 0.7, and our the issue was raised based on 0.6.171 . I will test and update the bug.